From 4fcc80c465dd11194e85eb2e2f52c7234646a20d Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Fri, 5 Sep 2014 19:49:23 -0400
Subject: [ethernet] bnx2x: save RAM in kdump kernel by disabling TPA

Message-id: <1409946615-5862-19-git-send-email-mschmidt@redhat.com>
Patchwork-id: 91559
O-Subject: [RHEL7.1 PATCH 18/70] bnx2x: save RAM in kdump kernel by disabling TPA
Bugzilla: 1059554 1089402
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1059554
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1089402

Upstream commit:
commit 94d9de3cf7b7902253e65d7aef30f2db051c1ac6
Author: Michal Schmidt <mschmidt@redhat.com>
Date:   Tue Feb 25 16:04:26 2014 +0100

    bnx2x: save RAM in kdump kernel by disabling TPA

    When running in a kdump kernel, disable TPA. This saves memory, which
    tends to be scarce in kdump.

    TPA, being a receive acceleration, is unlikely to be useful for kdump,
    whose purpose is to send the memory image out.

    This saves additional 5 MB in the kdump environment.

    Signed-off-by: Michal Schmidt <mschmidt@redhat.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_main.c b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_main.c
index 28af059..0a72486 100644
--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_main.c
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_main.c
@@ -11804,6 +11804,8 @@ static int bnx2x_init_bp(struct bnx2x *bp)
 
  bp->disable_tpa = disable_tpa;
  bp->disable_tpa |= IS_MF_STORAGE_SD(bp) || IS_MF_FCOE_AFEX(bp);
+ /* Reduce memory usage in kdump environment by disabling TPA */
+ bp->disable_tpa |= reset_devices;
 
  /* Set TPA flags */
  if (bp->disable_tpa) {
-- 
1.7.1